system/jlib/jtrace.cpp

```diff
@@ -791,6 +791,16 @@ class CSpan : public CInterfaceOf<ISpan>
         recordError(SpanError(msg.str(), e->errorCode(), spanFailed, escapedScope));
     };
 
+    virtual const char * queryTraceId() const override
```

Comment:
One things I noticed.  The trace id could only be stored in the server span, and all other spans could call parent->querySpan().  Definitely NOT something to do in this PR.

Reply:
Agreed.